### PR TITLE
Constructor for MR image data from MR acquisition data

### LINKED ISSUE #933: MR Image from MR Acquisition Data
Generate an empty image from MR Acquisition data. 
This is required for example in setting up an acquisition model without having to perform any reconstructions beforehand.

This addresses #824 on the MR side.

### PR DESCRIPTION
**Pull request by johannesmayer**
Fixes #933

The following operation is performed:
* for each consistent subsets of k-space a `CFImage` is created and appended. 
* the header is populated based on the first acquisition in each subset

While there are fields in the ISMRMRD XML header of the whole rawdata file comprising all acquisitions, the image headers are populated from the `Acquisition` object to include the geometry information that can vary between different subsets of k-space, for example for different 2D slices.
It is assumed that the `Acquisition` objects in each subset contain the same header information.

### CODE REVIEW

File: src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h [hunk 1]
```diff
@@ -876,7 +876,7 @@ namespace sirf {
 
 		GadgetronImagesVector() : images_()
 		{}
-		GadgetronImagesVector(const MRAcquisitionData& ad);
+		GadgetronImagesVector(const MRAcquisitionData& ad, const bool coil_resolved=false);
```
Comment (KrisThielemans) [inline src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h:879]: some doxygen doc please

File: src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h [hunk 2]
```diff
@@ -1058,6 +1058,8 @@ namespace sirf {
     public:
 
         CoilSensitivitiesVector() : GadgetronImagesVector(){}
+		CoilSensitivitiesVector(MRAcquisitionData& ad, const bool coil_resolved=true) :
+			GadgetronImagesVector(ad, coil_resolved){}
```
Comment (KrisThielemans) [inline src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h:1062]: would it make more sense to have
```suggestion
		CoilSensitivitiesVector(MRAcquisitionData& ad) :
			GadgetronImagesVector(ad, /*coil_resolved=*/true){}
```

**Review by KrisThielemans — CHANGES_REQUESTED** [review]
Obviously I have no idea if this is correct, but you know what you're doing so I trust you!

some tiny cosmetics therefore only. Do check the copyright statements as well, and add something to CHANGES.md

**Comment by KrisThielemans** [comment]
Please addd a test for set-up acquisition model (and maybe even basic recon)

File: src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h (hunk 2 above)
Comment (johannesmayer) [reply src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h]: done

File: src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h (hunk 1 above)
Comment (johannesmayer) [reply src/xGadgetron/cGadgetron/include/sirf/Gadgetron/gadgetron_data_containers.h:895]: done

**Review by KrisThielemans — APPROVED** [review]
ready to merge from my perspective. @johannesmayer ?

**Comment by johannesmayer** [comment]
I have found that in the special case of an empty `AcquisitionData` object an exception will be thrown.
So there is the option to
* leave it that way, and say if you don't have any acquired data it makes no sense to set up images from them.
* add some changes to use the k-space dimensions defined in the header to make images, but these won't contain geometry information? 

Which would you prefer?

**Comment by KrisThielemans** [comment]
I think best to use option 1 (throw an exception). Better safe than sorry.

Of course, this will mean we need to be able to create an `AcquisitionData` from scratch...

**Comment by KrisThielemans** [comment]
let us know if this needs any more changes then or not. I'd prefer to merge it soon such that it can be used in the SIRF-Exercises

**Comment by johannesmayer** [comment]
Ok then let's go with option 1, and I think it is good to merge.

**Pull request merged**